src/index.ts

```diff
@@ -1590,7 +1589,23 @@ export class BigQuery extends common.Service {
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
       options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+
+      // table#getRows uses listTableData endpoint, which is a faster method
+      // to read rows of results.
+
+      job!.getQueryResults(options, (err, rows) => {
+        if (!err) {
+          if (rows!.length !== 0) {
+            const datasetId = job!.metadata.configuration.query.destinationTable
+            .datasetId;
+            const tableId = job!.metadata.configuration.query.destinationTable
+              .tableId;
+            const dataset = this.dataset(datasetId);
+            const table = dataset.table(tableId);
```

Comment:
We're probably going to need to cache the `destinationTable` in case users don't want to `autoPaginate` the results